### DISCUSSION TITLE
Canonical Project Structure 🙏

### DISCUSSION THREAD

**Discussion by PierreMauger** [JeanPhilippeKernel/RendererEngine#310] [Ideas]
https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p1204r0.html

I recommend this paper about modern C++ project structure, separating components inside folders and keeping cpp and hpp files side by side. As the linked document explain, it helps navigate the project and keep a cleaner include strategy.

While its technical side is well executed, this project's structure could be improved, encouraging other people to contribute.

Have a great day.

**Comment by JeanPhilippeKernel** [comment]
Hello @PierreMauger  

Thanks a lot for this request and the doc paper reference ... that's awesome!
 I'll take time to review it and will propose a draft PR ASAP.

In case it's something you would like to lead I would be happy review your PR too 🙂

Once again Thanks a lot for bringing my attention to this, and sorry for late reply as it seems I missed this discussion

**Comment by JeanPhilippeKernel** [comment]
Hey @PierreMauger -

I'm closing this discussion as we shipped the canonical project structure months ago -
thanks once again for you proposal ! 

looking forward to your next contribution to Zodiac Engine ! :)
